flexibleengine/resource_flexibleengine_antiddos_v1.go

```diff
@@ -193,44 +198,40 @@ func resourceAntiDdosV1Delete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error deleting AntiDdos: %s", err)
 	}
 
-	time.Sleep(3 * time.Minute)
 	d.SetId("")
 	return nil
 }
 
-func waitForAntiDdosActive(antiddosClient *golangsdk.ServiceClient, antiddosId string) resource.StateRefreshFunc {
+func waitForAntiDdosStatus(antiddosClient *golangsdk.ServiceClient, antiddosId string) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		s, err := antiddos.Get(antiddosClient, antiddosId).Extract()
+		s, err := antiddos.GetStatus(antiddosClient, antiddosId).Extract()
 		if err != nil {
 			return nil, "", err
 		}
 
-		return s, "normal", nil
+		return s, s.Status, nil
 	}
 }
 
-func waitForAntiDdosDelete(antiddosClient *golangsdk.ServiceClient, antiddosId string) resource.StateRefreshFunc {
-	return func() (interface{}, string, error) {
+// checkNotConfig checks the error returned from the API
+func checkNotConfig(d *schema.ResourceData, err error, msg string) error {
 
-		r, err := antiddos.Get(antiddosClient, antiddosId).Extract()
-		if err != nil {
-			if _, ok := err.(golangsdk.ErrDefault403); ok {
-				log.Printf("[INFO] Successfully deleted antiddos %s", antiddosId)
-				return r, "notConfig", nil
-			}
-			return r, "normal", err
-		}
+	errResp, nErr := ParseErrorMsg(err.(golangsdk.ErrDefault400).Body)
```

Comment:
```suggestion
        if sdkErr, ok := err.(golangsdk.ErrDefault403); ok {
	        errResp, nErr := ParseErrorMsg(sdkErr.Body)
```

`comma ok` is recommended

Reply:
How to know the HTTP return code of the API?
400/403?

Reply:
you can get more details from https://github.com/chnsz/golangsdk/blob/master/provider_client.go#L366

Reply:
Thanks but how to know the HTTP return code of errors 10000016 and 10001020 in AntiDDos API ?

Reply:
`errResp.ErrorCode` is the error code

Reply:
Done

```
make testacc TEST='./flexibleengine' TESTARGS='-run TestAccAntiDdosV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./flexibleengine -v -run TestAccAntiDdosV1_basic -timeout 720m
=== RUN   TestAccAntiDdosV1_basic
--- PASS: TestAccAntiDdosV1_basic (221.36s)
PASS
ok      github.com/FlexibleEngineCloud/terraform-provider-flexibleengine/flexibleengine 222.011s
```